File: core/knowledge/ontology_restraint_engine.py
from __future__ import annotations

from typing import Any, Dict

from core.knowledge.speculative_ontology_engine import suppress_speculative_ontology_edge


def restrain_ontology_edge(edge: Dict[str, Any]) -> Dict[str, Any]:
    humbled = suppress_speculative_ontology_edge(edge)
    return {
        **humbled,
        "restraint": {"expansion_allowed": not humbled.get("unsupported", {}).get("edge", True)},
        "lineage": edge.get("lineage", {}),
    }

File: src/knowledge/ontologyRestraintEngine.ts
/**
 * Converted from Python: core/knowledge/ontology_restraint_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { suppressSpeculativeOntologyEdge } from "./speculativeOntologyEngine.js";

export function restrainOntologyEdge(edge: any): any {
  var humbled: any = suppressSpeculativeOntologyEdge(edge);
  return {...(humbled), "restraint": {"expansion_allowed": !py.truthy(py.get(py.get(humbled, "unsupported", {}), "edge", true))}, "lineage": py.get(edge, "lineage", {})};
}
export { suppressSpeculativeOntologyEdge };
